parlai/mturk/core/react_server/dev/app.jsx

```diff
@@ -308,7 +312,35 @@ class StaticApp extends React.Component {
   }
 
   onValidData(valid, response_data) {
-    this.setState({task_done: valid, response_data: response_data});
+    let all_response_data = this.state.response_data;
+    let show_next_task_button = false;
+    let task_done = true;
+    all_response_data[this.state.current_subtask_index] = response_data;
+    if (this.state.current_subtask_index < this.state.num_subtasks - 1) {
+      show_next_task_button = true;
+      task_done = false;
+    }
+    this.setState(
+      {
+        show_next_task_button: show_next_task_button,
+        subtask_done: valid,
+        task_done: task_done,
+        response_data: all_response_data,
+      },
+      this.render
+    );
+  }
+
+  nextButtonCallback() {
+    let next_subtask_index = this.state.current_subtask_index + 1;
+    this.setState(
+      {
+        current_subtask_index: next_subtask_index,
+        task_data: Object.assign(this.state.task_data, this.state.all_tasks_data[next_subtask_index]),
+        subtask_done: false,
+      },
+      this.reset
```

Comment:
Similarly I'm not sure what reset is doing here, the state change in theory should be sufficient to handle this